### PR TITLE
HHH-12278

### PR DESCRIPTION
**Pull request by benoitbesson**
This pull request includes : 
- the full change for classes AnnotationBinder and IdGeneratorInterpreterImpl
- additional classes to delegate on in package org.hibernate.cfg.schema
- test classes in org.hibernate.cfg.annotations.schema illustrating usage

I did not find how to check test results and automate assertions. When running tests with Gradle, I see a lot of error traces not related to this test. Maybe I need a little help... 

I hope this code could be enough to discuss on the proposal.

see https://hibernate.atlassian.net/browse/HHH-12278

### CODE REVIEW

**Comment by vladmihalcea** [comment]
The `SchemaNamingStrategyTest` fails with a NPE.

File: hibernate-core/src/main/java/org/hibernate/cfg/schema/SchemaNamingProviderLocator.java [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+ *
+ * TODO : integrate into class AnnotationSettings as "official" configuration. <br>
+ * TODO (later) : migrate into a non static approach... (complex)
+ *
+ * @author Benoit Besson
+ */
+public class SchemaNamingProviderLocator {
+
+	/** static instance to delegate on. */
+	private static SchemaNamingProvider instance;
```
Comment (vladmihalcea) [inline hibernate-core/src/main/java/org/hibernate/cfg/schema/SchemaNamingProviderLocator.java]: A static variable without any thread safety blocks is risky. Why not use a local variable and wrap it inside an object that confined to the caller.

File: hibernate-core/src/test/java/org/hibernate/cfg/annotations/schema/SchemaNamingStrategyTest.java [hunk 2]
```diff
@@ -0,0 +1,115 @@
...
+			schemaExport.setDelimiter(";");
+			schemaExport.setHaltOnError(true);
+			schemaExport.setImportFiles(null);
+			File outputFile = File.createTempFile("schema-screate", "sql");
+			schemaExport.setOutputFile(outputFile.getAbsolutePath());
+
+			// write to stdout and file
+			schemaExport.execute(EnumSet.of(TargetType.SCRIPT, TargetType.STDOUT), Action.CREATE, metadata);
+
+			// TODO : add assertion for expected file content
```
Comment (vladmihalcea) [inline hibernate-core/src/test/java/org/hibernate/cfg/annotations/schema/SchemaNamingStrategyTest.java]: The  asserts are a must have.

Comment (benoitbesson) [reply hibernate-core/src/test/java/org/hibernate/cfg/annotations/schema/SchemaNamingStrategyTest.java]: I agree.

I made this first proposal to discuss on functional usage without making the code too complex to understand. After discussing on how this change could be proposed, we could discuss on how making the good asserts.

Several approches may be done:  testing the generated schema file content, and/or testing metadata or other asserts. Can you help me deciding the good assert(s) to be added ?

**Comment by sebersole** [comment]
We decided to not do this - the issue was rejected, I just forgot to close the PR

**Pull request closed without merging**